### PR TITLE
INT-72: Payone optimisations

### PR DESCRIPTION
**Pull request by aleksey-kotsuba**
- Developer(s): @aleksey-kotsuba

- Ticket: https://spryker.atlassian.net/browse/INT-78

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Payone                | major                 |                       |

-----------------------------------------

##### Change log

Improvements

- Introduced partial capture feature.
- Introduced state machine examples with partial operations for CreditCard and OnlineTransfer methods.
- Adjusted CreditCard frontend integration to use card type from Payone.
- Adjusted tables definition with performance improvements (indexes, column types).

### CODE REVIEW

File: src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-credit-card/payone-credit-card.ts [hunk 1]
```diff
@@ -139,7 +162,10 @@ export default class PayoneCreditCard extends Component {
 
     get isCurrentPaymentMethod(): boolean {
         const currentPaymentMethodInput = <HTMLInputElement>document.querySelector(this.currentPaymentMethodSelector);
-        return currentPaymentMethodInput.value === CURRENT_PAYMENT_METHOD;
+
+        return currentPaymentMethodInput?.value
+            ? currentPaymentMethodInput.value === CURRENT_PAYMENT_METHOD
+            : undefined;
```
Comment (andrew-maslov) [inline src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-credit-card/payone-credit-card.ts:168]: plz use `false` instead `undefined`

File: src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-credit-card/payone-credit-card.ts [hunk 2]
```diff
@@ -87,17 +100,19 @@ export default class PayoneCreditCard extends Component {
...
-    protected onCardTypeChange(event: Event): void {
-        const inputType = <HTMLInputElement>event.currentTarget;
-        this.setCardType(inputType.value);
-    }
-
     protected onSubmit(event: Event): void {
-        if (!this.isCurrentPaymentMethod || this.isPaymentValid) {
+        event.preventDefault();
+
+        if (typeof this.isCurrentPaymentMethod === 'undefined') {
```
Comment (andrew-maslov) [inline src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-credit-card/payone-credit-card.ts:106]:
```suggestion
        if (this.isCurrentPaymentMethod) {
```

File: src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-credit-card/payone-credit-card.twig [hunk 3]
```diff
@@ -28,41 +28,43 @@
...
         </div>
 
-        <div class="form__field col col--sm-12">
+        <div class="form__field col col--sm-12 col--lg-6">
             {{ form_row(data.form.payoneCreditCard.cardholder, {
-                label: 'Payone_credit_card.cardholder'
+                label: 'Payone_credit_card.cardholder',
+                attr: {
+                    placeholder: 'checkout.step.payment.cardholder_name' | trans,
+                }
```
Comment (andrew-maslov) [inline src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-credit-card/payone-credit-card.twig:42]:
```suggestion
                },
```

File: src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-credit-card/payone-credit-card.ts [hunk 4]
```diff
@@ -139,7 +159,10 @@ export default class PayoneCreditCard extends Component {
 
     get isCurrentPaymentMethod(): boolean {
         const currentPaymentMethodInput = <HTMLInputElement>document.querySelector(this.currentPaymentMethodSelector);
-        return currentPaymentMethodInput.value === CURRENT_PAYMENT_METHOD;
+
+        return currentPaymentMethodInput?.value
+            ? currentPaymentMethodInput.value === CURRENT_PAYMENT_METHOD
+            : undefined;
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-credit-card/payone-credit-card.ts:165]: Suggest ti use `null` instead.

File: src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-credit-card/payone-credit-card.ts [hunk 5]
```diff
@@ -139,7 +159,10 @@ export default class PayoneCreditCard extends Component {
 
     get isCurrentPaymentMethod(): boolean {
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payone/Theme/default/components/molecules/payone-credit-card/payone-credit-card.ts:160]: Have you checked it with linters? Please add change return type to `bollean | null`.

**Pull request merged**